src/include/miopen/solver/mlir_common.hpp

```diff
@@ -41,10 +41,31 @@ std::string ConstructBuildOptions(const ConvolutionContext& ctx,
                                   bool is_xdlops,
                                   int kernel_id = 0);
 
+template <typename T>
 std::string ConstructBuildOptions(const ConvolutionContext& ctx,
-                                  const std::string& config,
+                                  const T& perf_config,
                                   bool is_xdlops,
-                                  int kernel_id = 0);
+                                  int kernel_id = 0)
+{
+    std::ostringstream mlir_handle;
+
+    bool isHeuristicRequest = false;
+    if(perf_config == T::GetHeuristicInitRequest())
+    {
+        isHeuristicRequest = true;
+    }
+
+    // Library does heuristic initialization when no perf_config
+    // is specified
+    // clang-format off
+    mlir_handle
+        << ConstructBuildOptions(ctx, is_xdlops, kernel_id)
+        << (isHeuristicRequest? "" : " --perf_config " + perf_config.ToString());
+    // clang-format on
+
+    return mlir_handle.str();
```

Comment:
There are missing parentheses around `" --perf_config " + perf_config.ToString()`. I would also refactor the code. Finally:
```suggestion
    std::ostringstream options = ConstructBuildOptions(ctx, is_xdlops, kernel_id);
    if(!(perf_config == T::GetHeuristicInitRequest()))
        options << " --perf_config " + perf_config.ToString();
    return options.str();
```

Reply:
This does look cleaner, thanks for your suggestion!